libclc/amdgcn-amdhsa/libspirv/atomic/atomic_add.cl

```diff
@@ -29,8 +29,13 @@ AMDGPU_ATOMIC(_Z18__spirv_AtomicIAdd, unsigned long, m, __hip_atomic_fetch_add)
     return __hip_atomic_fetch_add(p, val, memory_order, atomic_scope);                                     \
   }
 
-AMDGPU_ATOMIC_FP32_ADD_IMPL(global, U3AS1, 1, AMDGPU_ARCH_BETWEEN(9010, 10000),
-                            __builtin_amdgcn_global_atomic_fadd_f32)
+// Global AS atomics can be unsafe for malloc shared atomics, so should be opt
+// in
+AMDGPU_ATOMIC_FP32_ADD_IMPL(
+    global, U3AS1, 1,
+    __oclc_amdgpu_reflect("AMDGPU_OCLC_UNSAFE_FP_ATOMICS") &&
+        AMDGPU_ARCH_BETWEEN(9010, 10000),
```

Comment:
What happens if you swap the left and right side of the conditional here? Does it mean we can avoid extra LLVM passes from running?

Reply:
Good idea. It won't stop the pass from running but it will make it run faster if I swap order. Will do so now

Reply:
Change made.